src/daemon.rs

```diff
@@ -197,3 +199,34 @@ impl fmt::Display for NotifyState {
         }
     }
 }
+
+/// Perform some basic sanity checks against state entries.
+fn sanity_check_state_entries(state: &[NotifyState]) -> Result<(), SdError> {
+    for (index, entry) in state.iter().enumerate() {
+        match entry {
+            NotifyState::Fdname(ref name) => validate_fdname(name),
+            _ => Ok(()),
+        }
+        .map_err(|e| format!("invalid notify state entry #{}: {}", index, e))?;
```

Comment:
Can't we add a debug representation of the entry itself instead of its index?

Reply:
Mmh, I'm not a fan of adding a Debug entry in the middle of an error message; it is up to lower levels to add more details to pinpoint what's wrong. Currently a full error would look like this:
```
invalid notify state entry #0: invalid character ':' in fdname 'foo:bar'
```

If the index here does not carry useful information, we can even drop it. The nested error should already provide enough details anyway to identify the problem.

Reply:
Yes, that's the general idea: intermediate layers add some context about the flowpath, while the leaf error provide details about the fault.